### ISSUE TITLE
EventGrid Trigger for Functions does not work with EventTriggerAttribute in VS Tooling

### ISSUE THREAD

**Issue by ChristianWeyer** [Azure/azure-functions-eventgrid-extension#16]
When using the sample from here: https://docs.microsoft.com/en-us/azure/event-grid/resize-images-on-storage-blob-upload-event
everything works as expected - the event from a blog storage blob upload is fired.

However, when creating a new Azure Function from scratch in VS (not a .csx. but a compiled one) and add a reference to Microsoft.Azure.WebJobs.Extensions.EventGrid (1.0.0.-beta2) it compiles correctly, but after publishing to Functions in Azure, nothing happens when I add a file to the blob storage container.

No error, nothing.

I can see and verify that the subscription is in place - via:
az eventgrid resource event-subscription list --resource-name <storageaccount> --provider-namespace Microsoft.Storage --resource-type storageAccounts

Any ideas what could be wrong?

Thanks!

**Comment by drub0y** [comment]
Sooo, this is where working with compiled Azure Functions is a bit messy and unintuitive and... well, not at all how .NET traditionally works honestly. It comes down to a problem with .NET assembly versioning/binding...

*TL;DR;* - downgrade your Microsoft.Azure.WebJobs.Extensions.EventGrid to `1.0.0.-beta1` and try again.

The Microsoft.Azure.WebJobs.Extensions.EventGrid extension is actually baked into the Azure Functions Host runtime and the version that is baked in is actually `1.0.0-beta1`. If you create your VS project with `1.0.0-beta2` it will compile just fine, but if you try and run it within the host you are likely going to run into assembly versioning issues and discovery of your EventGrid based function will fail. I _believe_ something cryptic about this will be logged by the host... check your logs. I know for sure you will see something logged if you run the Azure Function locally with the local Azure Function Host debugging experience.

Bigger picture, be are this doesn't just affect the EventGrid assembly, it affects all assemblies the host has baked in and you won't really know if you're affected until you integration test your function as the compiler has no way to even warn you about this since it's a specific runtime/hosting issue.

**Comment by ChristianWeyer** [comment]
Hmmm... this didn't help. Thanks for the idea, tough! :-)

Just nothing happens. As if the subscription was not there. But it is.

Is there some sample/walkthrough to follow with Microsoft.Azure.WebJobs.Extensions.EventGrid to make sure things are really working?
Thanks!

**Comment by ChristianWeyer** [comment]
Or is there a recommended way to debug such kind of scenarios?

**Comment by drub0y** [comment]
Oh that's annoying. Well, def stick with that `beta1` version of the package while we figure this out. So what happens when you run it locally (i.e. F5)? As I said, the host outputs some pretty good diagnostic details that can hopefully lead us to an answer.

**Comment by ChristianWeyer** [comment]
Thanks!
I registered the subscription via the portal when the function is deployed to Azure.
Not sure how to trigger the function when debugging locally...

**Comment by drub0y** [comment]
Ah, so, yes... integration testing an actual event notification from Event Grid locally _is_ a little convoluted right now, but possible using a function proxy and something like nGrok to expose your locally running function to the cloud. Let's table that for a second though. 

What I want to make sure is that your function is even being detected and activated by the host run time. For that, a simple F5 should do and you should see a bunch of output that indicates that the function was detected, the webhook was started, etc. If you can do that and then paste the output to a gist and share a link to that gist here I can take a look at it.

**Comment by ChristianWeyer** [comment]
Ah, OK :-)

Nope, it seems it is not detected...

`[09.11.2017 20:39:07] Reading host configuration file 'C:\sources\thinktecture-research\CW\azure-homeautomation\HomeAutomation-Functions\bin\Debug\net461\host.json'
[09.11.2017 20:39:07] Host configuration file read:
[09.11.2017 20:39:07] // Set functionTimeout to 10 minutes
[09.11.2017 20:39:07] {
[09.11.2017 20:39:07]     "functionTimeout": "00:10:00"
[09.11.2017 20:39:07] }
Listening on http://localhost:7071/
Hit CTRL-C to exit...
[09.11.2017 20:39:07] Loaded custom extension 'BotFrameworkConfiguration'
[09.11.2017 20:39:07] Loaded custom extension 'SendGridConfiguration'
[09.11.2017 20:39:07] Loaded custom extension 'EventGridExtensionConfig'
[09.11.2017 20:39:08] Generating 5 job function(s)
[09.11.2017 20:39:08] Starting Host (HostId=desktop8ho2g1r-211019493, Version=1.0.11351.0, ProcessId=20212, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
[09.11.2017 20:39:08] Found the following functions:
[09.11.2017 20:39:08] HomeAutomationFunctions.Functions.AlertNotificationByHttp.Run
[09.11.2017 20:39:08] HomeAutomationFunctions.Functions.AlertNotificationByStorage.Run
[09.11.2017 20:39:08] HomeAutomationFunctions.Functions.ControlCommand.Run
[09.11.2017 20:39:08] HomeAutomationFunctions.Functions.GetBuildingInfo.Run
[09.11.2017 20:39:08] HomeAutomationFunctions.Functions.GetDeviceState.Run
[09.11.2017 20:39:08]
[09.11.2017 20:39:08] Job host started
[09.11.2017 20:39:08] Host lock lease acquired by instance ID '0000000000000000000000003FFB7B4A'.
[09.11.2017 20:39:08] Executing HTTP request: {
[09.11.2017 20:39:08]   "requestId": null,
[09.11.2017 20:39:08]   "method": "GET",
[09.11.2017 20:39:08]   "uri": "/"
[09.11.2017 20:39:08] }
[09.11.2017 20:39:08] Executed HTTP request: {
[09.11.2017 20:39:08]   "requestId": "4efb30b5-786d-4b64-9c78-200ce7cbe932",
[09.11.2017 20:39:08]   "method": "GET",
[09.11.2017 20:39:08]   "uri": "/",
[09.11.2017 20:39:08]   "authorizationLevel": "Anonymous"
[09.11.2017 20:39:08] }
[09.11.2017 20:39:08] Response details: {
[09.11.2017 20:39:08]   "requestId": "4efb30b5-786d-4b64-9c78-200ce7cbe932",
[09.11.2017 20:39:08]   "status": "OK"
[09.11.2017 20:39:08] }

Http Functions:

        alertnotificationhttp: http://localhost:7071/api/alertnotificationhttp

        controlcommand: http://localhost:7071/api/control/{hubDeviceId}/{targetDevice}/{command}

        getbuildinginfo: http://localhost:7071/api/buildinginfo/{hubDeviceId}

        getdevicestate: http://localhost:7071/api/state/{hubDeviceId}/{targetDevice}

Debugger listening on [::]:5858`

**Comment by drub0y** [comment]
Welll... that's certainly interesting, isn't it? Alright, that's good at least as it sets us in a direction

So let me first step back and check something with you. What version of the tooling are you using? Version `1.0.6` came out recently and I just realized I didn't have that version installed yet and was still using `1.0.4`. VS just suggested I update, so I did, and then I went to check to see what version of the EventGrid assembly they're including and they actually did bump to version `-beta2` now. So, if you're on 1.0.6 (which I think you are based on the host version info included in the output) I have to throw a mea-culpa out there for that original guidance. Can you re-update to `-beta2` and run the same test again and see if/how the host output changes?

**Comment by ChristianWeyer** [comment]
This is the JSON being generated, BTW:

`{
  "generatedBy": "Microsoft.NET.Sdk.Functions.Generator-1.0.6",
  "configurationSource": "attributes",
  "bindings": [
    {
      "type": "eventGridTrigger",
      "name": "eventGridEvent"
    }
  ],
  "disabled": false,
  "scriptFile": "../bin/HomeAutomation-Functions.dll",
  "entryPoint": "HomeAutomationFunctions.Functions.AlertNotificationByStorage.Run"
}`

**Comment by ChristianWeyer** [comment]
I am using 1.0.6.
Just did update to -beta 2. 

'
[09.11.2017 21:19:13] Host configuration file read:
[09.11.2017 21:19:13] // Set functionTimeout to 10 minutes
[09.11.2017 21:19:13] {
[09.11.2017 21:19:13]     "functionTimeout": "00:10:00"
[09.11.2017 21:19:13] }
Listening on http://localhost:7071/
Hit CTRL-C to exit...
[09.11.2017 21:19:13] Loaded custom extension 'BotFrameworkConfiguration'
[09.11.2017 21:19:13] Loaded custom extension 'SendGridConfiguration'
[09.11.2017 21:19:13] Loaded custom extension 'EventGridExtensionConfig'
[09.11.2017 21:19:14] Generating 5 job function(s)
[09.11.2017 21:19:14] Starting Host (HostId=desktop8ho2g1r-211019493, Version=1.0.11351.0, ProcessId=17812, Debug=False, ConsecutiveErrors=0, StartupCount=1, FunctionsExtensionVersion=)
[09.11.2017 21:19:14] Found the following functions:
[09.11.2017 21:19:14] HomeAutomationFunctions.Functions.AlertNotificationByHttp.Run
[09.11.2017 21:19:14] HomeAutomationFunctions.Functions.AlertNotificationByStorage.Run
[09.11.2017 21:19:14] HomeAutomationFunctions.Functions.ControlCommand.Run
[09.11.2017 21:19:14] HomeAutomationFunctions.Functions.GetBuildingInfo.Run
[09.11.2017 21:19:14] HomeAutomationFunctions.Functions.GetDeviceState.Run
[09.11.2017 21:19:14]
[09.11.2017 21:19:14] Job host started
[09.11.2017 21:19:15] Executing HTTP request: {
[09.11.2017 21:19:15]   "requestId": null,
[09.11.2017 21:19:15]   "method": "GET",
[09.11.2017 21:19:15]   "uri": "/"
[09.11.2017 21:19:15] }
[09.11.2017 21:19:15] Executed HTTP request: {
[09.11.2017 21:19:15]   "requestId": "ebb3aada-2893-4def-a6e2-a59380c89909",
[09.11.2017 21:19:15]   "method": "GET",
[09.11.2017 21:19:15]   "uri": "/",
[09.11.2017 21:19:15]   "authorizationLevel": "Anonymous"
[09.11.2017 21:19:15] }
[09.11.2017 21:19:15] Response details: {
[09.11.2017 21:19:15]   "requestId": "ebb3aada-2893-4def-a6e2-a59380c89909",
[09.11.2017 21:19:15]   "status": "OK"
[09.11.2017 21:19:15] }

Http Functions:

        alertnotificationhttp: http://localhost:7071/api/alertnotificationhttp

        controlcommand: http://localhost:7071/api/control/{hubDeviceId}/{targetDevice}/{command}

        getbuildinginfo: http://localhost:7071/api/buildinginfo/{hubDeviceId}

        getdevicestate: http://localhost:7071/api/state/{hubDeviceId}/{targetDevice}

Debugger listening on [::]:5858
[09.11.2017 21:19:17] Host lock lease acquired by instance ID '0000000000000000000000003FFB7B4A'.
'

**Comment by ChristianWeyer** [comment]
That is my csproj:
'
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
    <RootNamespace>HomeAutomationFunctions</RootNamespace>
  </PropertyGroup>
  <ItemGroup>    
    <PackageReference Include="IdentityModel" Version="2.12.0" />    
    <PackageReference Include="Microsoft.Azure.Devices" Version="1.4.1" />    
    <PackageReference Include="Microsoft.Azure.NotificationHubs" Version="1.0.9" />    
    <PackageReference Include="Microsoft.Azure.WebJobs" Version="2.1.0-beta4" />    
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DocumentDB" Version="1.1.0-beta4" />    
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.EventGrid" Version="1.0.0-beta2" />    
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.NotificationHubs" Version="1.1.0-beta4" />    
    <PackageReference Include="Microsoft.Azure.WebJobs.ServiceBus" Version="2.1.0-beta4" />    
    <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="1.0.6" />    
    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="5.2.0-preview1-408290725" />
  </ItemGroup>
  <ItemGroup>
    <ProjectReference Include="..\HomeAutomation-Common\HomeAutomation-Common.csproj" />
  </ItemGroup>
  <ItemGroup>
    <Reference Include="Microsoft.CSharp" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
</Project>
'

**Comment by drub0y** [comment]
Ok, so actually, now that I see your `function.json` I see your function entryPoint is:

`HomeAutomationFunctions.Functions.AlertNotificationByStorage.Run`

And in your host output we see that it's discovering that guy:

> [09.11.2017 21:19:14] Found the following functions:
> [09.11.2017 21:19:14] HomeAutomationFunctions.Functions.AlertNotificationByHttp.Run
> **[09.11.2017 21:19:14] HomeAutomationFunctions.Functions.AlertNotificationByStorage.Run**
> [09.11.2017 21:19:14] HomeAutomationFunctions.Functions.ControlCommand.Run
> [09.11.2017 21:19:14] HomeAutomationFunctions.Functions.GetBuildingInfo.Run
> [09.11.2017 21:19:14] HomeAutomationFunctions.Functions.GetDeviceState.Run

That's the one, correct? If so I think we can rule out the discovery as being the problem now.

So, let's move on to the next "bug" of the EventGrid extension... do you have a `FunctionName` attribute on your function and, if so, is the name you gave it the same name as the actual .NET method? There is currently a bug that will cause it to only respond to web callbacks based on the actual .NET method name. Your .NET method name is clearly just `"Run"`, so you need to make sure you have also specified `[FunctionName("Run")]`. If this is a problem and you fix it, the next step would be going and fixing the EventGrid Subscription you created. The `Full Endpoint URL` of the subscription will need to now have `?functionName=Run` in the querystring instead of whatever name was originally there.

**Comment by ChristianWeyer** [comment]
Ah, OK - makes sense.

This is my function:

```
public static class AlertNotificationByStorage
    {
        [FunctionName("alertnotificationstorage")]
        public static void Run(
            [EventGridTrigger]
            EventGridEvent eventGridEvent,

            TraceWriter log)
        {
            log.Info($"EventGridEvent" +
                $"\n\tId:{eventGridEvent.Id}" +
                $"\n\tTopic:{eventGridEvent.Topic}" +
                $"\n\tSubject:{eventGridEvent.Subject}" +
                $"\n\tType:{eventGridEvent.EventType}" +
                $"\n\tData:{JsonConvert.SerializeObject(eventGridEvent.Data)}");
        }
    }
```
and this is the subscription:
```
[
  {
    "destination": {
      "endpointBaseUrl": "https://tt-homeautomation-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig",
      "endpointType": "WebHook",
      "endpointUrl": null
    },
    "filter": {
      "includedEventTypes": [
        "All"
      ],
      "isSubjectCaseSensitive": null,
      "subjectBeginsWith": "/blobServices/default/containers/clientpackages/blobs/",
      "subjectEndsWith": ""
    },
    "id": "/subscriptions/b3a47414-171f-48e3-93f7-ab39117ae927/resourceGroups/Dev-CW/providers/Microsoft.Storage/StorageAccounts/tthomeautomationfiles/providers/Microsoft.EventGrid/eventSubscriptions/test2",
    "labels": [
      ""
    ],
    "name": "test2",
    "provisioningState": "Succeeded",
    "resourceGroup": "Dev-CW",
    "topic": "/subscriptions/b3a47414-171f-48e3-93f7-ab39117ae927/resourceGroups/dev-cw/providers/microsoft.storage/storageaccounts/tthomeautomationfiles",
    "type": "Microsoft.EventGrid/eventSubscriptions"
  }
]
```

I guess this is not OK, right...? ;-)

**Comment by drub0y** [comment]
Ok, so yeah, there's two problems:

First, definitely change that `FunctionName` attribute value to just be `"Run"`. Either that or change your .NET method name to be `AlertNotificationStorage` and match the case in the function name just for good measure. They need to match today. [This is a bug in the EventGrid extension that I've reported here](https://github.com/Azure/azure-functions-eventgrid-extension/issues/14).

Second, is this exactly what's in that Subscriber Endpoint right now???

>  https://tt-xxx-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig

No querystring at all? If so, yeah that's missing some info. Specifically there should be something that looks like this:

>  https://<myappname>-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig?functionName=Run&code=MnUDPLNpvtfw2OzalPyjqSr32lQERseiYao5WIgkmtOvcOg5WjRRkA==

Did you use the built in `Add Event Grid subscription` link button in the portal? I'm assuming not since I've never seen it leave off those required query string details.

So the `functionName` param is easy since you know it's gonna be either `Run` or `AlertNotificationStorage` if go with the more logical name. The `code` param on the other hand... you're not gonna be able to guess that or even get that any easy way (that I know of). That's generated by the webhook infrastructure that the EventGrid extension is piggy backing off of and, honestly, I haven't ever looked into how that value is generated. I do know it's based off of some data about your type/assembly, because it manages to stay constant across builds/deployments from what I've seen. Anyway, what I recommend is you delete your existing subscription and then go back to the function in the portal, click on the function itself and then click that `Add Event Grid subscription` link button I mentioned:

![snip_20171109234443](https://user-images.githubusercontent.com/284152/32647858-383a9aaa-c5a8-11e7-81ec-160d3a586333.png)

That will send the correct link, querystring and all, for the EventGrid webhook into the subscription creation flow for you and then you just need to configure the rest of the details. Once you've done that I think you'll start to see the events flowing in!

**Comment by ChristianWeyer** [comment]
Actually I did use that "Add Event Grid subscription" button... hm...

**Comment by ChristianWeyer** [comment]
My function now looks like this:
```
[FunctionName("alertnotificationstorage")]
        public static void alertnotificationstorage(
            [EventGridTrigger]
            EventGridEvent eventGridEvent,

            TraceWriter log)
        {
```

After publishing I get this error in the portal:
![image](https://user-images.githubusercontent.com/888718/32649193-73b949f4-c5f9-11e7-8e11-f80a7bd2f9a9.png)

Something is very wrong here :-)

**Comment by ChristianWeyer** [comment]
Are we sure that this command does return what we are looking for?
`az eventgrid resource event-subscription list --resource-name xxx --provider-namespace Microsoft.Storage --resource-type storageAccounts`

Because when I issue this command for the sample here: https://docs.microsoft.com/en-us/azure/event-grid/resize-images-on-storage-blob-upload-event

... I get this, which looks similar to the above output. And that sample works, the event is fired... hmm ...
```
[
  {
    "destination": {
      "endpointBaseUrl": "https://yyy.azurewebsites.net/admin/extensions/EventGridExtensionConfig",
      "endpointType": "WebHook",
      "endpointUrl": null
    },
    "filter": {
      "includedEventTypes": [
        "Microsoft.Storage.BlobCreated"
      ],
      "isSubjectCaseSensitive": null,
      "subjectBeginsWith": "/blobServices/default/containers/images/blobs",
      "subjectEndsWith": ""
    },
    "id": "/subscriptions/b3a47414-171f-48e3-93f7-ab39117ae927/resourceGroups/Dev-CW/providers/Microsoft.Storage/StorageAccounts/cwttimages/providers/Microsoft.EventGrid/eventSubscriptions/imageresizesub",
    "labels": [
      ""
    ],
    "name": "imageresizesub",
    "provisioningState": "Succeeded",
    "resourceGroup": "Dev-CW",
    "topic": "/subscriptions/b3a47414-171f-48e3-93f7-ab39117ae927/resourceGroups/dev-cw/providers/microsoft.storage/storageaccounts/xyxy",
    "type": "Microsoft.EventGrid/eventSubscriptions"
  }
]
```

**Comment by drub0y** [comment]
@ChristianWeyer congrats, you seem to have found yet another bug! I can confirm that listing the subscription out with the Azure CLI excludes the querystring that _I know_ is there. Hey, it's early beta... we can't be that mad... can we? 🤔 😆  

If you go click on the same subscription in the portal to open the editor and copy out the `Full Endpoint Url`, do you see those querystring params I mentioned on it? I'm going to assume they _are_ there considering you used the `Add Event Grid subscription` link button.

Ok, so that should put us back to the root problem where you had a .NET method named `Run` and that didn't match the `FunctionAttribute` name. If I'm following correctly, you changed that and republished, right? I think I saw you had posted a picture of a failure at one point, but that's not here in the thread any more so I think you worked through it. 

If you're still stuck after all this, can you, in your next reply, please paste the latest `function.json` and that `Full Endpoint Url` value? Basically we have to make sure that the `functionName` querystring param definitvely aligns with the function name (i.e. both should be `"AlertNotificationStorage"` now).

**Comment by ChristianWeyer** [comment]
Hey Drew - yeah, kind of feels like back in those days of NGWS/.NET Alphas... hehehe
So, do I get a shirt now...? 😂😇

OK.

0. Yeah, that image with an error: it seems that was intermittent... gone.

1. I do not see any way to list subscriptions in the portal - where exactly?

2. This is my function code:
```
namespace HomeAutomationFunctions.Functions
{
    public static class AlertNotificationByStorage
    {
        [FunctionName("AlertNotificationStorage")]
        public static void AlertNotificationStorage(
            [EventGridTrigger]
            EventGridEvent eventGridEvent,

            TraceWriter log)
        {
            log.Info($"EventGridEvent" +
                $"\n\tId:{eventGridEvent.Id}" +
                $"\n\tTopic:{eventGridEvent.Topic}" +
                $"\n\tSubject:{eventGridEvent.Subject}" +
                $"\n\tType:{eventGridEvent.EventType}" +
                $"\n\tData:{JsonConvert.SerializeObject(eventGridEvent.Data)}");
        }
    }
}
```

3. This is the function.json from the portal:
```
{
  "generatedBy": "Microsoft.NET.Sdk.Functions.Generator-1.0.6",
  "configurationSource": "attributes",
  "bindings": [
    {
      "type": "eventGridTrigger",
      "name": "eventGridEvent"
    }
  ],
  "disabled": false,
  "scriptFile": "../bin/HomeAutomation-Functions.dll",
  "entryPoint": "HomeAutomationFunctions.Functions.AlertNotificationByStorage.AlertNotificationStorage"
}
```

Thanks man!

**Comment by drub0y** [comment]
Here's how you get to the subscriptions in the portal:

Start by finding the Event Grid Topic resource and clicking on that. At the bottom of that blade is the list of subscriptions. Find your subscription and click on it and that will pop out the editing panel from the right hand side which will contain that `Full Endpoint URL` form field I've been talking about. That form field will have that little copy decorator on it that you can click to get the value or you just do what you would do with any old text box to view/copy the text.

![untitled](https://user-images.githubusercontent.com/284152/32675489-e8bb20f8-c60b-11e7-80f9-6904fb635501.png)

**Comment by ChristianWeyer** [comment]
Ah, well hidden. Thanks, found it.

`https://tt-xxx-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig?functionName=alertnotificationstorage&code=6xbYSupXhvZzd7Ltbw2yHvnBZu4x5LtVCGIlznhZl1SHV9bkiCNKvQ==`

**Comment by drub0y** [comment]
Awesome. Ok, so... there's the querystring, notice anything fishy 🐟?? The `functionName` param is all lower case, right? Guess what? It's case-sensitive. ☹️ I really couldn't believe it either, so I double checked by changing my test function here to not match on case and it didn't get invoked; changed it back and it worked again.

So, you can just go ahead hand edit it there to match case and click the `Save Changes` button and try to trigger it again! 🤞 🤞

**Comment by ChristianWeyer** [comment]
I tried both:
`https://tt-homeautomation-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig?functionName= AlertNotificationStorage&code=6xbYSupXhvZzd7Ltbw2yHvnBZu4x5LtVCGIlznhZl1SHV9bkiCNKvQ==`

and

`https://tt-homeautomation-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig?functionName= alertnotificationstorage&code=6xbYSupXhvZzd7Ltbw2yHvnBZu4x5LtVCGIlznhZl1SHV9bkiCNKvQ==`

Doesn't trigger.

**Comment by drub0y** [comment]
Err, are those exact values or did it just paste weird? Cause there's a space after the `functionName=`.

**Comment by ChristianWeyer** [comment]
Let's try that again:

`https://tt-xxx-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig?functionName=alertnotificationstorage&code=6xbYSupXhvZzd7Ltbw2yHvnBZu4x5LtVCGIlznhZl1SHV9bkiCNKvQ==`

and

`https://tt-xxx-functions.azurewebsites.net/admin/extensions/EventGridExtensionConfig?functionName=AlertNotificationStorage&code=6xbYSupXhvZzd7Ltbw2yHvnBZu4x5LtVCGIlznhZl1SHV9bkiCNKvQ==`

![image](https://user-images.githubusercontent.com/888718/32677685-aae997ce-c65f-11e7-8b5c-ad2b62983ce3.png)

Still no trigger.

**Comment by ChristianWeyer** [comment]
It is getting a bit more weird 😄

I put six files into the container - but only one fired... and only after many minutes.
```
2017-11-10T20:50:43  No new trace in the past 14 min(s).
2017-11-10T20:51:43  No new trace in the past 15 min(s).
2017-11-10T20:52:14.657 Function started (Id=fd676409-b28f-4470-b981-bec35df9f3d2)
2017-11-10T20:52:14.891 EventGridEvent
	Id:a74b72f5-001e-001b-2763-5a3d0a060424
	Topic:/subscriptions/b3a47414-171f-48e3-93f7-ab39117ae927/resourceGroups/Dev-CW/providers/Microsoft.Storage/storageAccounts/yyy
	Subject:/blobServices/default/containers/images/blobs/foo.xlsx
	Type:Microsoft.Storage.BlobCreated
	Data:{"api":"PutBlob","clientRequestId":"b2305ad0-c656-11e7-8a63-5b5f14aa0d50","requestId":"a74b72f5-001e-001b-2763-5a3d0a000000","eTag":"0x8D5287A976F212E","contentType":"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet","contentLength":58733,"blobType":"BlockBlob","url":"https://yyy.blob.core.windows.net/images/foo.xlsx","sequencer":"00000000000017330000000000130AC3","storageDiagnostics":{"batchId":"d51df04e-8255-4a98-b6d8-61c836c1711f"}}
2017-11-10T20:52:14.911 Function completed (Success, Id=fd676409-b28f-4470-b981-bec35df9f3d2, Duration=243ms)
2017-11-10T20:53:43  No new trace in the past 1 min(s).
2017-11-10T20:54:43  No new trace in the past 2 min(s).
2017-11-10T20:55:43  No new trace in the past 3 min(s).
```

The other files did not trigger.

**Comment by ChristianWeyer** [comment]
Just deleted the subscription again. Created a new one. And edited the functionName again.
Now it works.

😅
Tough stuff.

**Comment by drub0y** [comment]
Wowsers. 🤥

Well I'm glad you tried that 'cause I was running out of suggestions. 😆

**Comment by ChristianWeyer** [comment]
Thanks for all your help!
Hope this gets a bit more stable... soon-ish 🤓
